### PR TITLE
Can't Save a Database to a new File

### PR DESCRIPTION
**Pull request by zacmks**
Fixed the following issue: 

https://github.com/zacmks/jabref/issues/1

Also, update the maven repo, with secure http

### CODE REVIEW

**Comment by koppor** [comment]
Thank you for the PR. I think, our checkstyle check will result in errors. We are aware that checkstyle may be seen as annoying (https://github.com/JabRef/jabref/issues/2996). Our aim is to have the same code style over the whole project and thus, we decided to keep checkstyle in place. We currently have no howto hosted at the JabRef dev doc, but the howto at another project shows how to include checkstyle in IntelliJ: http://eclipse.github.io/winery/dev/config/IntelliJ%20IDEA/README.html.

File: build.gradle [hunk 1]
```diff
@@ -5,7 +5,7 @@ import org.gradle.internal.os.OperatingSystem
 buildscript {
     repositories {
         maven {
-            url 'http://maven.ej-technologies.com/repository'
+            url 'https://maven.ej-technologies.com/repository'
```
Comment (lynyus) [inline build.gradle]: This is a very good point, and @koppor and me don't see any downside for using https. Despite that, this has nothing to do with the rest this PR, so I would like you to open another PR for this.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Sun Jul 09 12:08:20 CEST 2017
+#Tue Jul 18 01:58:04 BRT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.0.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.0.1-all.zip
```
Comment (lynyus) [inline gradle/wrapper/gradle-wrapper.properties]: Please keep this file as before and adjust your IDE settings accordingly.

File: src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java [hunk 3]
```diff
@@ -336,12 +334,18 @@ public void saveAs(File file) throws Exception {
...
-        AutosaveManager.shutdown(context);
-        BackupManager.shutdown(context);
+        try {
+            final Path oldFile = context.getDatabasePath().get();
+            context.setDatabaseFile(oldFile.toFile());
+            //closing AutosaveManager and BackupManager for original library
+            AutosaveManager.shutdown(context);
+            BackupManager.shutdown(context);
+        } catch (NoSuchElementException e) {
+            LOGGER.info("Old file not found, just creating a new file", e);
```
Comment (lynyus) [inline src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java]: `context.getDatabasePath().get()` returns an `Optional` which you can check for being present/absent using `isPresent()`. Then you won't need a try-catch block.

Check out this page to learn the most important features of `Optional`s https://examples.javacodegeeks.com/core-java/util/optional/java-8-optional-example/

**Comment by zacmks** [comment]
Sorry guys, @koppor, @lynyus. Should be fixed on the latest commit. Not very used to all these tools, I'm used to use JIRA, GitLab, and formatting with IDE messed up with style, sorry. I was just doing some class stuff and found the issue (next time, I'll check more carefully the projects' requirements). And thank you very much for the tips!

**Comment by lynyus** [comment]
@zacmks Good work! I'll merge this now. Will you open the other pull request for the https maven repository?

**Pull request merged**